esy-lib/posix_getumask_patch.c

```diff
@@ -0,0 +1,46 @@
+#include <caml/mlvalues.h>
+#include <caml/memory.h>
+#include <caml/alloc.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+
+#if __APPLE__ || __linux__
```

Comment:
This will break on freebsd, but above all https://ocaml.org/api/Unix.html#VALumask

OCaml exposes a `umask` already.

Reply:
@EduardoRFS @ManasJayanth in node when we call `process.umask()` I think this is called https://github.com/nodejs/node/blob/e83c8aef4b787b531f23ad7abb498fe81db5ee83/src/node_process_methods.cc#L223-L237
There is no special handling for windows

Reply:
On Windows process.umask always returns 0, not sure why, but that seems to be the case. So it would be reasonable, which means we don't need any C code, just `Unix.umask`